org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/GradleWorkspaceTest.groovy

```diff
@@ -0,0 +1,50 @@
+package org.eclipse.buildship.core
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.runtime.NullProgressMonitor
+
+import org.eclipse.buildship.core.internal.CorePlugin
+import org.eclipse.buildship.core.internal.configuration.GradleProjectNature
+import org.eclipse.buildship.core.internal.test.fixtures.WorkspaceSpecification
+
+class GradleWorkspaceTest extends WorkspaceSpecification {
+
+    def "Cannot get reference to a null Gradle build"() {
+        when:
+        GradleCore.workspace.getBuild(null)
+
+        then:
+        thrown(NullPointerException)
+    }
+
+    def "Cannot get reference to Gradle build if target project is inaccessible"() {
+        setup:
+        IProject project = newClosedProject("GradleWorkspaceTest")
+
+        when:
+        GradleCore.workspace.getBuild(project)
+
+        then:
+        thrown(IllegalArgumentException)
+    }
+
+    def "Cannot get reference to Gradle build if target project is not a Gradle project"() {
```

Comment:
❌ Using an exception for this feels wrong, especially since this method would essentially be the only way to test whether something is indeed a Gradle project. Should probably return null instead or use an Optional return type.

Reply:
Replaced with `Optional`.